### PR TITLE
feat: Patch ecrecover

### PR DESCRIPTION
**Pull request by Bisht13**
Patch `recover_ecdsa` function to use `ecdsa-core` for SP1 acceleration for ECDSA recovery.

### CODE REVIEW

File: src/ecdsa/recovery.rs [hunk 1]
```diff
@@ -192,6 +192,19 @@ impl<C: Verification> Secp256k1<C> {
         msg: &Message,
         sig: &RecoverableSignature,
     ) -> Result<key::PublicKey, Error> {
+        cfg_if::cfg_if! {
+            if #[cfg(all(target_os = "zkvm", target_vendor = "succinct"))] {
+                let prehash: &[u8] = &msg.0;
+                let mut rev_sig = sig.0[..32].iter().rev().chain(sig.0[32..64].iter().rev()).cloned().collect::<Vec<u8>>();
```
Comment (ratankaliani) [inline src/ecdsa/recovery.rs]: Specify the endian-ness of the incoming data and make the logic easy to follow with comments.

```rust
let sig_bytes_be = sig.0
# In XYZ, it's LE, need to convert to this.
let sig_bytes_le = 
```

File: src/ecdsa/recovery.rs [hunk 2]
```diff
@@ -192,6 +192,19 @@ impl<C: Verification> Secp256k1<C> {
...
+                let signature = sp1_ecdsa::Signature::<k256::Secp256k1>::from_slice(&rev_sig).unwrap();
+                let recovery_id = sp1_ecdsa::RecoveryId::from_byte(sig.0[64]).unwrap();
```
Comment (ratankaliani) [inline src/ecdsa/recovery.rs:219]: Here specify where sig.0[64] comes from

File: Cargo.toml [hunk 3]
```diff
@@ -54,6 +55,11 @@ bincode = "1.3.3"
 wasm-bindgen-test = "0.3"
 getrandom = { version = "0.2", features = ["js"] }
 
+[target.'cfg(all(target_os = "zkvm", target_vendor = "succinct"))'.dependencies]
+sp1-ecdsa = { path = "/home/adityabisht64/signatures/ecdsa", package = "ecdsa", features = ["verifying", "alloc"] }
```
Comment (ratankaliani) [inline Cargo.toml]: Point this to https://github.com/sp1-patches/signatures on the `patch-ecdsa-v0.16.9`

File: src/ecdsa/recovery.rs (hunk 1 above)
Comment (Bisht13) [reply src/ecdsa/recovery.rs]: Explained

File: src/ecdsa/recovery.rs (hunk 2 above)
Comment (Bisht13) [reply src/ecdsa/recovery.rs:219]: Explained

**Pull request closed without merging**